evoting/lib/transaction.go

```diff
@@ -0,0 +1,117 @@
+package lib
+
+import (
+	"errors"
+	"time"
+
+	"github.com/dedis/onet"
+	"github.com/dedis/onet/network"
+
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/dedis/cothority"
+	"github.com/dedis/cothority/skipchain"
+)
+
+func init() {
+	network.RegisterMessage(&Transaction{})
+}
+
+// TransactionVerifierID identifes the core transaction verification function.
+var TransactionVerifierID = skipchain.VerifierID(uuid.NewV5(uuid.NamespaceURL, ""))
+
+// TransactionVerifiers is a list of accepted skipchain verification functions.
+var TransactionVerifiers = []skipchain.VerifierID{TransactionVerifierID}
+
+// Transaction is the sole data structure withing the blocks of an election
+// skipchain, it holds all the other containers.
+type Transaction struct {
+	Election *Election
+	Ballot   *Ballot
+	Mix      *Mix
+	Partial  *Partial
+
+	User      uint32
+	Signature []byte
+}
+
+// UnmarshalTransaction decodes a data blob to a transaction structure.
+func UnmarshalTransaction(data []byte) *Transaction {
```

Comment:
We should ban the use of `network.Marshal` - it doesn't play nice with non-onet code :( Can you replace that with regular `protobuf.Encode` and `protobuf.Decode` calls?

Reply:
Fixed.